### PR TITLE
Stop embedding bitcode for iOS in tool

### LINKED ISSUE #107883: Remove bitcode support from the flutter tool
Once Xcode 14 is released, remove bitcode from the `flutter` tool and integration tests.

> Starting with Xcode 14, bitcode is no longer required for watchOS and tvOS applications, and the App Store no longer accepts bitcode submissions from Xcode 14.

Remove `-fembed-bitcode` and `-fembed-bitcode-marker` and `bcsymbolmap` logic.
https://github.com/flutter/flutter/blob/dffddf00a22e77122d67f4503efef16a46c18ec2/packages/flutter_tools/lib/src/commands/build_ios_framework.dart#L155-L157

We never turned bitcode on by default in our template, but just in case this should wait on the migration at https://github.com/flutter/flutter/issues/111776

See https://github.com/flutter/flutter/issues/15288 and https://github.com/flutter/flutter/pull/36471 when bitcode was introduced.

### PR DESCRIPTION
**Pull request by jmagman**
> Starting with Xcode 14, bitcode is no longer required for watchOS and tvOS applications, and the App Store no longer accepts bitcode submissions from Xcode 14.

https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes

- Stop embedding bitcode and markers in App.framework.  
- Always strip bitcode from Flutter.framework in release (until https://github.com/flutter/flutter/issues/107884 is complete).
- Disable bitcode for Flutter plugins in Podfile so they do not link on stripped Flutter.framework
- Stop embedding `bcsymbolmap` bitcode symbol maps as part of `flutter build ios-framework`

Fixes https://github.com/flutter/flutter/issues/107883

https://github.com/flutter/flutter/pull/112828 project migration should be merged first.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/build_system/targets/ios_test.dart [hunk 1]
```diff
@@ -696,12 +678,19 @@ void main() {
...
 
       processManager.addCommands(<FakeCommand>[
-        copyPhysicalFrameworkCommand,
+        FakeCommand(command: <String>[
+          'rsync',
+          '-av',
+          '--delete',
+          '--filter',
+          '- .DS_Store/',
+          'Artifact.flutterFramework.TargetPlatform.ios.release.EnvironmentType.physical',
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/build_system/targets/ios_test.dart:691]: `fails when bitcode strip fails` test will only pass during `ReleaseUnpackIOS` since stripping will now only happen on release, not dependent on `kBitcodeFlag`.

File: packages/flutter_tools/test/host_cross_arch.shard/ios_content_validation_test.dart [hunk 2]
```diff
@@ -358,3 +357,46 @@ void main() {
      timeout: const Timeout(Duration(minutes: 7))
   );
 }
+
+bool _containsBitcode(String pathToBinary, ProcessManager processManager) {
```
Comment (jmagman) [inline packages/flutter_tools/test/host_cross_arch.shard/ios_content_validation_test.dart:361]: This was moved from the now-deleted `darwin_common.dart`, bitcode isn't a thing on macOS, so it's not shared. I don't know why I ever added the check to the macOS validation test.

File: packages/flutter_tools/lib/src/build_system/targets/ios.dart [hunk 3]
```diff
@@ -298,7 +293,9 @@ abstract class UnpackIOS extends Target {
       throw Exception('Binary $frameworkBinaryPath does not exist, cannot thin');
     }
     _thinFramework(environment, frameworkBinaryPath, archs);
-    _bitcodeStripFramework(environment, frameworkBinaryPath);
+    if (buildMode == BuildMode.release) {
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/build_system/targets/ios.dart:296]: why this change?

**Review by christopherfujino — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/build_system/targets/ios.dart (hunk 3 above)
Comment (jmagman) [reply packages/flutter_tools/lib/src/build_system/targets/ios.dart:296]: (Currently until we remove it) the release version of `Flutter.framework` has a giant bitcode blob, and now we always want to strip it out.

I deleted the logic passing the Xcode build setting `ENABLE_BITCODE` down into `assemble` as `kBitcodeFlag`.  It would only be true if `ENABLE_BITCODE` was set, AND it was an "install" command, meaning it was archiving.  
https://github.com/flutter/flutter/blob/b22db680089a2170e13a52f571bc14eec546c75e/packages/flutter_tools/bin/xcode_backend.dart#L336-L339

This was the check in the target that was deleted:
https://github.com/flutter/flutter/blob/723b82e4f0f3652dfaffb6e9bbc714e069da2976/packages/flutter_tools/lib/src/build_system/targets/ios.dart#L377-L380

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/build_system/targets/ios.dart:296]: Ahh, makes sense. Thanks!

File: packages/flutter_tools/lib/src/build_system/targets/ios.dart [hunk 4]
```diff
@@ -373,16 +370,14 @@ abstract class UnpackIOS extends Target {
...
   void _bitcodeStripFramework(Environment environment, String frameworkBinaryPath) {
-    if (environment.defines[kBitcodeFlag] == 'true') {
-      return;
-    }
     final ProcessResult stripResult = environment.processManager.runSync(<String>[
       'xcrun',
       'bitcode_strip',
       frameworkBinaryPath,
-      '-m', // leave the bitcode marker.
+      '-r', // Delete the bitcode segment.
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/build_system/targets/ios.dart:380]: Just made this change, totally delete it instead of leaving the marker:
```
OPTIONS
       -r     Remove the __LLVM bitcode segment entirely.

       -m     Remove the bitcode from the __LLVM segment, leaving behind a
              marker.
```
